src/components/HURUmap/Map/index.js

```diff
@@ -30,23 +26,30 @@ function Map({
   geometries,
   geography,
   tileLayers,
+  preferredChildren,
+  featuredGeographies,
   ...props
 }) {
   const classes = useStyles(props);
   const [selectedBoundary, setSelectedBoundary] = useState(null);
 
-  const getSelectedBoundary = (level, geoms) => {
-    const preferredChildren = preferredChildrenObj[level];
-    if (!preferredChildren) return null;
+  const getSelectedBoundary = useCallback(
+    (level, geoms) => {
+      const preferredLevelChildren = preferredChildren[level];
+      if (!preferredLevelChildren) return null;
 
-    const availableLevels = preferredChildren.filter((l) => geoms.children[l]);
+      const availableLevels = preferredLevelChildren.filter(
+        (l) => geoms.children[l]
+      );
 
-    if (availableLevels.length > 0) {
-      const preferredLevel = availableLevels[0];
-      return geoms.children[preferredLevel];
-    }
-    return null;
-  };
+      if (availableLevels.length > 0) {
+        const preferredLevel = availableLevels[0];
```

Comment:
If all we need is `availableLevels[0]`, then wouldn't `.find` be a better method that `.filter`?